diff --git a/node_modules/jsxgraph/src/3d/circle3d.js b/node_modules/jsxgraph/src/3d/circle3d.js
index 2e9438a..2d4d980 100644
--- a/node_modules/jsxgraph/src/3d/circle3d.js
+++ b/node_modules/jsxgraph/src/3d/circle3d.js
@@ -264,7 +264,7 @@ JXG.createCircle3D = function (board, parents, attributes) {
     el = new JXG.Circle3D(view, center, normal, radius, attr);
 
     // update scene tree
-    el.center.addChild(el);
+    el.center.addParents(el);
     el.addChild(el.curve);
 
     el.update();
diff --git a/node_modules/jsxgraph/src/3d/view3d.js b/node_modules/jsxgraph/src/3d/view3d.js
index b911e87..a23fa2f 100644
--- a/node_modules/jsxgraph/src/3d/view3d.js
+++ b/node_modules/jsxgraph/src/3d/view3d.js
@@ -1832,20 +1832,20 @@ JXG.createView3D = function (board, parents, attributes) {
 
     attr = Type.copyAttributes(attributes, board.options, 'view3d');
     view = new JXG.View3D(board, parents, attr);
-    view.defaultAxes = view.create('axes3d', parents, attributes);
+    view.defaultAxes = view.create('axes3d', parents, attr);
 
     x = coords[0];
     y = coords[1];
     w = size[0];
     h = size[1];
 
-    attr_az = Type.copyAttributes(attributes, board.options, 'view3d', 'az', 'slider');
+    attr_az = Type.copyAttributes(attr, board.options, 'view3d', 'az', 'slider');
     attr_az.name = 'az';
 
-    attr_el = Type.copyAttributes(attributes, board.options, 'view3d', 'el', 'slider');
+    attr_el = Type.copyAttributes(attr, board.options, 'view3d', 'el', 'slider');
     attr_el.name = 'el';
 
-    attr_bank = Type.copyAttributes(attributes, board.options, 'view3d', 'bank', 'slider');
+    attr_bank = Type.copyAttributes(attr, board.options, 'view3d', 'bank', 'slider');
     attr_bank.name = 'bank';
 
     /**
