diff --git a/node_modules/jsxgraph/src/jsxgraph.js b/node_modules/jsxgraph/src/jsxgraph.js
index ec09c7e..3cd93f0 100644
--- a/node_modules/jsxgraph/src/jsxgraph.js
+++ b/node_modules/jsxgraph/src/jsxgraph.js
@@ -343,6 +343,10 @@ JXG.JSXGraph = {
             // Size of HTML div.
             // If zero, the size is set to a small value to avoid
             // division by zero.
+
+			// Obsidian does not load the div before this is called so I have to override dimensions to not be zero
+			dimensions.width = "1";
+			dimensions.height = "1";
             w = Math.max(parseInt(dimensions.width, 10), Mat.eps);
             h = Math.max(parseInt(dimensions.height, 10), Mat.eps);
 
